homeassistant/components/switchbot_cloud/binary_sensor.py

```diff
@@ -0,0 +1,107 @@
+"""Support for SwitchBot Cloud binary sensors."""
+
+from dataclasses import dataclass
+
+from switchbot_api import Device, SwitchBotAPI
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import SwitchbotCloudData
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+from .entity import SwitchBotCloudEntity
+
+
+@dataclass(frozen=True)
+class SwitchBotCloudBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes a Switchbot Cloud binary sensor."""
+
+    # Value or values to consider binary sensor to be "on"
+    on_value: bool | float | int | str | set[bool | float | int | str] = True
```

Comment:
why do we have so many types when we only use 2 in real life? Alternatively you can also just create a value_fn so we can implement it differently per enttiy description

Reply:
thanks, good catch!